File: src/main/java/net/matees/settings/global/listeners/PlayerJoinListener.java
package net.matees.settings.global.listeners;

import org.bukkit.GameMode;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

import net.matees.settings.global.GlobalSettings;

public class PlayerJoinListener implements Listener {
    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        event.getPlayer()
                .setAllowFlight((boolean) GlobalSettings.getInstance().getSetting("Enable Flight").getSetting());
        if (event.getPlayer().getGameMode() == GameMode.CREATIVE)
            event.getPlayer().setAllowFlight(true);
    }
}

File: src/main/java/net/matees/settings/global/listeners/PlayerJoinListener.kt
package net.matees.settings.global.listeners

import net.matees.settings.global.GlobalSettings
import org.bukkit.GameMode
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class PlayerJoinListener : Listener {
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        event.player.allowFlight =
            GlobalSettings.Companion.instance.getSetting("Enable Flight")?.setting as Boolean
        if (event.player.gameMode == GameMode.CREATIVE) event.player.allowFlight = true
    }
}
